solr/core/src/java/org/apache/solr/core/MMapDirectoryFactory.java

```diff
@@ -63,7 +63,7 @@ protected Directory create(String path, LockFactory lockFactory, DirContext dirC
       throws IOException {
     MMapDirectory mapDirectory = new MMapDirectory(Path.of(path), lockFactory, maxChunk);
     try {
-      mapDirectory.setUseUnmap(unmapHack);
+      System.setProperty(MMapDirectory.ENABLE_UNMAP_HACK_SYSPROP, String.valueOf(unmapHack));
```

Comment:
Okay, will remove the `unmapHack` references.

Reply:
I'd still add a warning message.

The default was always "true". If somebody changed it from the default, I'd log a warning with  with the URL to the Lucene documentation as stated before.

> If we want to be nicer to users - then logging a warning is a valid option.

This would bring both together.

Reply:
you mean something like this?
```
@Override 
  protected Directory create(String path, LockFactory lockFactory, DirContext dirContext)
      throws IOException {
    MMapDirectory mapDirectory = new MMapDirectory(Path.of(path), lockFactory, maxChunk);
    log.warn("to disable the unmap hack globally, see this Lucene documentation" +
            " https://lucene.apache.org/core/9_5_0/core/org/apache/lucene/store/MMapDirectory.html#ENABLE_UNMAP_HACK_SYSPROP");
    mapDirectory.setPreload(preload);
    return mapDirectory;
  }
```

Reply:
No not in create().
On init() just check for existence of directory property "unmap", if it exists show exactly this warning. Maybe add something "Per directory unmapping is no longer supported with Lucene 9.5. ...."
If user does not specify anything in the XML file, no warning is printed.

Reply:
Like `if (params.get("unmap") != null)` (this is a good check for existence in SolrParams.

Reply:
Do it like that:

```java
public class MMapDirectoryFactory extends StandardDirectoryFactory {
  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
  boolean preload;
  private long maxChunk;

  @Override
  public void init(NamedList<?> args) {
    super.init(args);
    SolrParams params = args.toSolrParams();
    maxChunk = params.getLong("maxChunkSize", MMapDirectory.DEFAULT_MAX_CHUNK_SIZE);
    if (maxChunk <= 0) {
      throw new IllegalArgumentException("maxChunk must be greater than 0");
    }
    if (params.get("unmap") != null) {
      log.warn("It is no longer possible to configure unmapping of index files on DirectoryFactory level in solrconfig.xml.");
      log.warn("To disable unmapping, pass -Dorg.apache.lucene.store.MMapDirectory.enableUnmapHack=false on Solr's command line.");
    }
    preload = params.getBool("preload", false); // default turn-off
  }

  @Override
  protected Directory create(String path, LockFactory lockFactory, DirContext dirContext)
      throws IOException {
    MMapDirectory mapDirectory = new MMapDirectory(Path.of(path), lockFactory, maxChunk);
    mapDirectory.setPreload(preload);
    return mapDirectory;
  }
}
```

Reply:
Sure, will do that. Thanks